pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,143 @@ $title: Use custom JavaScript in AMP pages
 
-## Scope
+### Replace amp-bind and amp-list
 
-Any DOM elements the custom JavaScript files wish to interact with must be wrapped inside the `<amp-script>` component tags. This includes other AMP components. The `<amp-script>` component considers `document.body` to be the `<amp-script>` element and not the document's `<body>` element.
+AMP has an extensive system of [actions and events](../learn/amp-actions-and-events.md) that allow you to handle browser events by taking common actions on the DOM or by mutating the state of AMP components. It also includes interactive components like [`amp-bind`](../../examples/documentation/amp-bind.html) and [`amp-list`](../../../documentation/components/reference/amp-list.md). If you're new to AMP and you're comfortable with JavaScript, it may be easier to simply use `amp-script`. For more elaborate interactions, `amp-script` will [likely be easier](#amp-script---or-amp-bind-and-amp-list).
 
-If you were to call `document.body.appendChild(document.createElement('span'))` within the script imported into an `<amp-script>` element in the following document:
+### Handle server data
+
+AMP allows you to retrieve server data using `amp-list` and to format it using [`amp-mustache`](../../examples/documentation/amp-mustache.html). In cases where mustache templates are insufficient, `amp-script` can fetch the data, format it, and inject the formatted data into the DOM. If you need to massage server data before sending it to `amp-mustache`, an `amp-script` function can be the data source for `amp-list`. See [the documentation](../../../documentation/components/reference/amp-list.md?format=websites#initialization-from-amp-state) for details and a code sample.
+
+### Introduce new capabilities
+
+You can use `amp-script` to leverage areas of the Web API and DOM API that aren't currently accessible to AMP components, or to use these APIs in ways that AMP components don't support. For example, `amp-script` supports `WebSockets` ([example](https://amp.dev/documentation/examples/components/amp-script/#using-a-websocket-for-live-updates)), `localStorage`, and `Canvas`. It supports browser events that AMP doesn't pass to traditional components. And since `amp-script` provides access to the `navigator` object, you can retrieve information about [the user's browser](https://amp.dev/documentation/examples/components/amp-script/#detecting-the-operating-system) or [preferred language](https://amp.dev/documentation/examples/components/amp-script/#personalization).
```

Comment:
"It supports browser events that AMP doesn't pass to traditional components." like what?

Reply:
making this more specific. @samouri , @kristoferbaxter is there a good way to characterize the events that WorkerDom *doesn't* support? Looking [at the code](https://github.com/ampproject/worker-dom/blob/main/src/worker-thread/Event.ts), it seems to me that WorkerDom supports whatever events come to it, and it includes a lot of standard event object properties... but of course a lot of events would never be seen in `amp-script` because we don't support the whole DOM API.

Right now I've replaced this with the still rather vague phrase "a wide variety of browser events".

Reply:
I don't know offhand. I'll let Kris respond / if he's not sure either i'll research

Reply:
that's ok. I can also keep this vague. But we should make a note to document the browser events, and event properties/methods, that `amp-script` does support.

It's so just vast...

https://developer.mozilla.org/en-US/docs/Web/API/Event